libobs/obs-video-gpu-encode.c

```diff
@@ -125,10 +134,36 @@ static void *gpu_encode_thread(struct obs_core_video_mix *video)
 			else
 				next_key++;
 
-			success = encoder->info.encode_texture(
-				encoder->context.data, tf.handle,
-				encoder->cur_pts, lock_key, &next_key, &pkt,
-				&received);
+			profile_start(gpu_encode_frame_name);
+			if (encoder->info.encode_texture2) {
+				union {
+					struct encoder_texture tex;
+					/* MSVC complains about
+					   offsetof(..., tex[3]) */
+					char dummy[offsetof(struct encoder_texture,
+							    tex) +
+						   sizeof(struct gs_texture *) *
+							   3];
+				} u = {0};
```

Comment:
What's this about?

Reply:
The most difficult code to read is always your past self... I believe msvc blew up when trying to zero initialize this struct, so this added a byte-array of sufficient size. Lets remove it and see the error in CI.

Reply:
It seems like even if MSVC hates it that it would be saner to just `memset` the struct rather than... this.

Reply:
Actually i think this was left over from when we used a VLA instead of a sized array for gs_texture pointers. At least its working with msvc so monstrosity removed. Thanks.